core-kotlin-modules/core-kotlin-concurrency-3/src/main/java/com/baeldung/comparedelays/CompareDelays.kt

```diff
@@ -0,0 +1,45 @@
+package com.baeldung.comparedelays
+
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.*
+
+
+val lock = Object()
+
+fun main() {
+    runBlocking(Dispatchers.Default) {
+        launch(Dispatchers.IO) {
+            testWaitThread1()
+        }
+        launch(Dispatchers.IO) {
+            testWaitThread2()
+        }
+    }
+}
+
+fun testWaitThread1() = synchronized(lock) {
+    println("Print first")
+    lock.wait()
+}
+
+fun testWaitThread2() = synchronized(lock) {
+    println("Print second")
+    lock.notify()
+}
```

Comment:
One last nit here, let's print after the wait, otherwise it can be hard to see that the wait did anything at all.

```suggestion
fun testWaitThread1() = synchronized(lock) {
        lock.wait()
        println("Print first")
}

fun testWaitThread2() = synchronized(lock) {
    println("Print second")
    lock.notify()
}
```